src/bind.rs

```diff
@@ -232,27 +220,24 @@ where
     B: tower_h2::Body + Send + 'static,
     <B::Data as ::bytes::IntoBuf>::Buf: Send,
 {
-    /// Binds the "inner" layers of the stack.
-    ///
-    /// This binds a service stack that comprises the client for that individual
-    /// endpoint. It will have to be rebuilt if the TLS configuration changes.
+    /// Binds the innermost layers of the stack with a TLS configuration.
     ///
-    /// This includes:
-    /// + Reconnects
-    /// + URI normalization
-    /// + HTTP sensors
+    /// A reconnecting HTTP client is established with the given endpont,
+    /// protocol, and TLS configuration.
     ///
-    /// When the TLS client configuration is invalidated, this function will
-    /// be called again to bind a new stack.
-    fn bind_reconnect_stack(
+    /// This client is instrumented with metrics.
+    fn bind_with_tls(
```

Comment:
TIOLI: I think it would be ideal if the name of this function (and the returned stack type) indicated that this function is responsible for setting up both TLS and reconnects. But, on the other hand, I think `bind_with_tls_and_reconnects` is much too wordy, so I'm not sure if there's a better option here...